client/my-sites/upgrades/cart/cart-plan-ad.jsx

```diff
@@ -1,33 +1,40 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 import page from 'page';
+import React, { Component } from 'react';
 
 /**
  * Internal dependencies
  */
 import CartAd from './cart-ad';
 import { cartItems } from 'lib/cart-values';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isDomainOnlySite } from 'state/selectors';
 import { isPlan } from 'lib/products-values';
 import * as upgradesActions from 'lib/upgrades/actions';
 import { PLAN_PREMIUM } from 'lib/plans/constants';
 
-const CartPlanAd = React.createClass( {
+class CartPlanAd extends Component {
 	addToCartAndRedirect( event ) {
 		event.preventDefault();
 		upgradesActions.addItem( cartItems.premiumPlan( PLAN_PREMIUM, { isFreeTrial: false } ) );
 		page( '/checkout/' + this.props.selectedSite.slug );
-	},
+	}
+
 	shouldDisplayAd() {
 		const { cart, selectedSite } = this.props;
 
-		return cart.hasLoadedFromServer &&
+		return ! this.props.isDomainOnlySite &&
```

Comment:
Super minor but since we are already destructuring props we could do the same for this one:

```
const { cart, isDomainOnlySite, selectedSite } = this.props;

return ! sDomainOnlySite &&
```